libraries/Microsoft.Bot.Builder.AI.QnA/QnAMakerRecognizer.cs

```diff
@@ -158,7 +158,11 @@ public override async Task<RecognizerResult> RecognizeAsync(DialogContext dialog
             List<Metadata> filters = new List<Metadata>();
             if (IncludeDialogNameInMetadata.GetValue(dialogContext.State))
             {
-                filters.Add(new Metadata() { Name = "dialogName", Value = dialogContext.ActiveDialog.Id });
+                filters.Add(new Metadata()
+                {
+                    Name = "dialogName",
+                    Value = dialogContext.ActiveDialog.Id.EndsWith(".dialog") ? dialogContext.ActiveDialog.Id.Substring(0, dialogContext.ActiveDialog.Id.LastIndexOf(".")) : dialogContext.ActiveDialog.Id
+                });
```

Comment:
Why?

Reply:
More accurately, the reason this is the dialog id is because it is the dialog id.  You should NOT be messing with this.